packages/website/blog/2024-05-27-announcing-typescript-eslint-v8-beta.mdx

```diff
@@ -0,0 +1,352 @@
+
+In typescript-eslint v8, we've made two changes to `RuleCreator`:
+
+- Rule `meta.docs` by default only allows the properties defined in ESLint's [Custom Rules > Rule Structure docs](https://eslint.org/docs/latest/extend/custom-rules#rule-structure): `description` and `url`
+- `RuleCreator` allows an optional type parameter to specify additional allowed properties
+
+For example, this rule includes the common `meta.docs.recommended` property as a `boolean`:
+
+```ts
+interface MyPluginDocs {
+  recommended: boolean;
+}
+
+const createRule = ESLintUtils.RuleCreator<MyPluginDocs>(
+  name => `https://example.com/rule/${name}`,
+);
+
+createRule({
+  // ...
+  meta: {
+    docs: {
+      description: '...',
+      recommended: true,
+    },
+    // ...
+  },
+});
+```
+
+See [feat(utils): allow specifying additional rule meta.docs in RuleCreator](https://github.com/typescript-eslint/typescript-eslint/pull/9025) for more details.
+
+### Support for multi-pass fixes in `RuleTester`
+
+The `RuleTester` provided by `@typescript-eslint/rule-tester` is a fork of ESLint's `RuleTester`.
+One limitation of the original `RuleTester` is that it is not possible to verify the individual applied fixes when a rule provides multiple rounds of fixes.
+[The original `RuleTester` applies only the first fix](https://eslint.org/docs/latest/integrate/nodejs-api#testing-fixes) when there is conflict between two fixes.
+
+In typescript-eslint v8, `RuleTester` tries to apply all possible fixes for each test case.
+
+If your rule tests had some test cases that required multi-pass fixes, you will see some test failures.
+To fix these failures, provide an array of strings for `output` which specifies the output after each fix pass.
+
+```ts
+import { RuleTester } from '@typescript-eslint/rule-tester';
+import rule from '../src/rules/my-rule.ts';
+
+const ruleTester = new RuleTester();
+
+ruleTester.run('my-rule', rule, {
+  valid: [
+    /* ... */
+  ],
+  invalid: [
+    {
+      code: 'const a = 1;',
+      // Remove the line with string form of `output`
+      output: 'const b = 1;',
+      // Add the line with array form of `output`
+      output: ['const b = 1;', 'const c = 1;'],
+      errors: [
+        /* ... */
+      ],
+    },
+  ],
+});
+```
+
+See [[rule-tester] support multipass fixes](https://github.com/typescript-eslint/typescript-eslint/issues/8554) for more details.
+
+### Other Developer-Facing Breaking Changes
+
+- ⏳ [Repo: Rule [options] parameter should be non-nullable if defaultOptions exists](https://github.com/typescript-eslint/typescript-eslint/issues/5439)
+- [feat(parser): always enable comment, loc, range, tokens](https://github.com/typescript-eslint/typescript-eslint/pull/8617)
+  - If you were manually calling `@typescript-eslint/parser` functions, those options are no longer necessary to provide
+- [feat(rule-tester): support multipass fixes](https://github.com/typescript-eslint/typescript-eslint/pull/8883)
```

Comment:
```suggestion
```

Reply:
followup for https://github.com/typescript-eslint/typescript-eslint/pull/9022#discussion_r1613236762 :smile: